prow/plugins/project/project.go

```diff
@@ -0,0 +1,270 @@
+	ListTeamMembers(id int, role string) ([]github.TeamMember, error)
+	GetRepos(org string, isUser bool) ([]Repo, error)
+	GetRepoProjects(owner, repo string) ([]Project, error)
+	GetOrgProjects(org string) ([]Project, error)
+	GetProjectColumns(projectID int) ([]ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard ProjectCard) (*ProjectCard, error)
+	GetProjectCard(projectID int, number int) (ProjectCard, error)
+	MoveProjectCard(projectCardID int, newColumnID int) error
+	DeleteProjectCard(projectCardID int) error
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	msgForTeam := func(team plugins.Project) string {
+		return fmt.Sprintf(projectTeamMsg, team.MaintainersTeam, team.MaintainersID)
+	}
+
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The project plugin allows members of a configurable GitHub team to set the project and column on an issue or pull request.",
+		Config: func(repos []string) map[string]string {
+			configMap := make(map[string]string)
+			for _, repo := range repos {
+				team, exists := config.ProjectConfig[repo]
+				if exists {
+					configMap[repo] = msgForTeam(team)
+				}
+			}
+			configMap[""] = msgForTeam(config.ProjectConfig[""])
+			return configMap
+		}(enabledRepos),
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/project <board>, /project <board> <column>, or /project clear <board>",
+		Description: "Updates the project board and column for an issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Members of the project maintainers GitHub team can use the '/project' command.",
+		Examples:    []string{"/project 0.5.0", "/project 0.5.0 3999801", "/project clear 0.4.0"},
+	})
+	return pluginHelp, nil
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.PluginConfig.ProjectConfigMap)
+}
+
+func buildProjectMap(projects []github.Project) map[string]int {
+	m := make(map[string]int)
+	for _, project := range projects {
+		m[project.Name] = project.ID
+	}
+	return m
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, projectConfigMap map[string]plugins.ProjectConfig) error {
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	matches := projectRegex.FindStringSubmatch(e.Body)
+	if len(matches) != 2 && len(matches) != 3 {
+		return nil
+	}
+
+	proposedProject := matches[1]
+	proposedColumn := -1
+	if len(matches) == 3 && proposedProject != clearKeyword {
+		columnInt, err := strconv.Atoi(matches[2])
+		if err == nil {
+			proposedColumn = columnInt
+		}
+	}
+
+	// See if the current user is part of the team in order to set the project/column
+	projectConfig, exists := projectConfigMap[fmt.Sprintf("%s/%s", org, repo)]
```

Comment:
This type of resolution -- use `org/repo` or default -- could belong to the config package, see other examples there